owa/app/js/components/manage_patient_identifier_sources/existingIdentifierSources.jsx

```diff
@@ -0,0 +1,95 @@
+import React, {Component} from 'react';
+import {Table, Button, Modal, ModalHeader, ModalBody, ModalFooter} from 'reactstrap';
+import ViewSource from './viewSource';
+import ConfigureSource from './configureSource';
+
+export default class ExistingIdentifierSources extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      modal: false,
+      modalContent: '',
+      identifierSources: [{
+        identifierType: ['OpenMRS ID'],
+        sourceType: ['Local Identifier Generator'],
+        sourceName: ['New Source']
+      }]
+    };
+
+    this.toggle = this.toggle.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  //show the modal
+  toggle() {
+    this.setState({
+      modal: !this.state.modal
+    });
+  }
+
+  //handle clicks on the buttons
+  handleClick(event){
+    if(event.target.name === 'configureButton'){
+      this.setState({
+        modalContent: <ConfigureSource />
+      });
+    }else{
+      this.setState({
+        modalContent: <ViewSource />
+      });
+    }
+    this.toggle()
+  }
+
+  render() {
+
+    return (
+
+      <div className="mpisContainer">
+        <div className="mpisHeader">Existing Identifier Sources</div>
+        <Table size="sm" responsive className="mpisTable">
+          <thead>
+            <tr>
+              <th>Identifier Type</th>
+              <th>Source Type</th>
+              <th>Source Name</th>
+              <th>Actions</th>
+            </tr>
+          </thead>
+          <tbody>
+            {
+              Object.keys(this.state.identifierSources).map(index => {
+                return (
+                  <tr className="mpisRow" key={index}>
```

Comment:
Please what is `mpis` that I keep seeing everywhere? I guess it's something relating to the project title, right? Just a suggestion, please is there a way you can make the name/ids more readable?

Reply:
If the `mpis` is really something relating to the project name or title as I guessed, you may want to leave it the way it is

Reply:
Yes mpis stands for Manage Patient Identifier Sources, which is the page I am working on. I did that so that my CSS does not affect other pages. Do you have any suggestions on how to make the names more readable>

Reply:
Would managePatientIdentifierSources hurt? 😊